app_dart/lib/src/request_handlers/reset_prod_task.dart

```diff
@@ -55,20 +61,65 @@ class ResetProdTask extends ApiRequestHandler<Body> {
     final String? repo = requestData![repoParam] as String?;
     final String? sha = requestData![commitShaParam] as String?;
     final TokenInfo token = await tokenInfo(request!);
-    final String? taskName = requestData![builderParam] as String?;
+    final String? taskName = requestData![taskParam] as String?;
 
     RepositorySlug? slug;
-
     if (encodedKey != null && encodedKey.isNotEmpty) {
       // Check params required for dashboard.
       checkRequiredParameters(<String>[taskKeyParam]);
     } else {
       // Checks params required when this API is called with curl.
-      checkRequiredParameters(<String>[commitShaParam, builderParam, repoParam]);
+      checkRequiredParameters(<String>[commitShaParam, taskParam, repoParam]);
       slug = RepositorySlug(owner, repo!);
       gitBranch ??= Config.defaultBranch(slug);
     }
 
+    final bool retryAll = taskName == 'all';
+
+    if (retryAll) {
+      final BuildStatusService buildStatusService = BuildStatusService(datastore);
+      final List<CommitStatus> statuses = await buildStatusService.retrieveCommitStatus(slug: slug!, limit: 5).toList();
+      final CommitStatus status = statuses.firstWhere((CommitStatus status) => status.commit.sha == sha);
+      final List<Future<void>> futures = <Future<void>>[];
+      for (final Task task in status.stages.first.tasks) {
```

Comment:
Umm does this mean the rerun will be on every target regardless of its success/failure status, or there is logic somewhere that checks to only rerun the failed tasks?

Reply:
See https://cs.opensource.google/flutter/cocoon/+/main:app_dart/lib/src/service/luci_build_service.dart;l=639

ignoreChecks: false will have LuciBuildService verify if the task should be rerun (i.e. it's in a failing state)

Reply:
thanks for explaining!

Reply:
Nit: logically it would be more clear if we loop through only failed tasks.